### ISSUE TITLE
[ng] http guide - Cannot find reflection information on BrowserClient exception

### ISSUE THREAD

**Issue by Janamou** [dart-lang/site-webdev#332]
Hi,

if I register the `BrowserClient` according to the guide here: https://webdev.dartlang.org/angular/guide/server-communication#!#http-client

e.g.: 
```dart
bootstrap(AppComponent, const [BrowserClient]);
```

I get the "Exception: Cannot find reflection information on BrowserClient" ([the issue](https://github.com/angular/angular.io/issues/2347)).

I think that you should explicitly mention, that until this problem is fixed, devs need to use the longer the version:

e.g.:
```dart
bootstrap(AppComponent, [provide(BrowserClient, useFactory: () => new BrowserClient(), deps: [])]);
```

P.S.: The example is ok:
https://github.com/angular-examples/server-communication/blob/master/web/main.dart

**Comment by chalin** [comment]
Sadly, the server-communication page has become out-of-date (on both the TS and Dart sides). The [tutorial already has the suggestion you have made](https://webdev.dartlang.org/angular/tutorial/toh-pt6#register-for-http-services).

I've opened #334 to address the need to update the server-communication page. I'll make the specific change mentioned in this issue sooner though.

**Comment by chalin** [comment]
@Janamou - by the way, thank you for the through issue descriptions. I've submitted a fix via #335.
